### PR TITLE
[FEE-524] Address dependabot issues

### PR DESCRIPTION
**Pull request by mdalpozzo**
- resolve vulnerable @babel/traverse dependency: https://github.com/appfolio/react-gears/security/dependabot/76
- resolve vlunerable http-cache-semantics dependency:https://github.com/appfolio/react-gears/security/dependabot/63

### CODE REVIEW

**Review by aaronmars — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -86,12 +86,12 @@
...
-    "@babel/core": "^7.17.8",
-    "@babel/preset-env": "^7.16.11",
+    "@babel/core": "^7.24.7",
+    "@babel/preset-env": "^7.24.7",
     "@babel/preset-react": "^7.16.7",
-    "@babel/preset-typescript": "^7.16.7",
+    "@babel/preset-typescript": "^7.24.7",
     "@chromatic-com/storybook": "^1.5.0",
-    "@jest/types": "^27.5.1",
+    "@jest/types": "^29.6.3",
```
Comment (davidacevedo) [inline package.json:94]: Shouldn't this be v28?

File: package.json [hunk 2]
```diff
@@ -160,7 +161,9 @@
   },
   "resolutions": {
     "@types/react": "^16.14.26",
-    "@types/react-dom": "^16.9.16"
+    "@types/react-dom": "^16.9.16",
+    "@babel/traverse": "^7.24.7",
```
Comment (davidacevedo) [inline package.json:165]: Why was this needed?

**Comment by davidacevedo** [comment]
I recommend following this document when fixing security vulnerabilities https://coda.io/d/Front-end-All-Not-official-yet_dg8nZlrMnpj/How-to-Address-Dependabot-Alerts_suw3B#_luMyy
You shouldn't need yarn resolutions.

File: package.json (hunk 2 above)
Comment (mdalpozzo) [reply package.json:165]: some transitive instances are lagging in @storybook/react-webpack5 and jest

Screenshot 2024-06-18 at 1 48 07 PM

File: package.json [hunk 3]
```diff
@@ -160,7 +161,9 @@
...
+    "http-cache-semantics": "^4.1.1"
```
[the remark this answers is not in the document]
Comment (mdalpozzo) [reply package.json:166]: Screenshot 2024-06-18 at 1 50 19 PM

File: package.json (hunk 1 above)
Comment (mdalpozzo) [reply package.json:94]: no i updated, jest to v29

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (davidacevedo) [reply package.json:166]: I'd recommend looking at the doc I sent as it should resolve the issue you are having. The `^` sign means you can update to the latest without it being forced with `resolutions`

[the remark this answers is not in the document]
Comment (davidacevedo) [reply package.json:166]: Usually `yarn up -R package-name` is enough

[the remark this answers is not in the document]
Comment (mdalpozzo) [reply package.json]: yeah good call

File: package.json (hunk 1 above)
Comment (davidacevedo) [reply package.json:94]: Ah, I got confused with `@types/jest`. Can you upgrade that to v29 as well. It seems like it's available https://www.npmjs.com/package/@types/jest

Comment (mdalpozzo) [reply package.json:94]: yes, thanks!

File: package.json [hunk 4]
```diff
@@ -86,12 +86,12 @@
   "devDependencies": {
     "@appfolio/eslint-config-appfolio-react": "^2.0.2",
     "@babel/cli": "^7.17.6",
-    "@babel/core": "^7.17.8",
-    "@babel/preset-env": "^7.16.11",
+    "@babel/core": "^7.24.7",
```
Comment (davidacevedo) [inline package.json:89]: So usually when you update 1 `@babel` package, you should update all of them. So you might also want to update `@babel/cli` and `@babel/preset-react`.

**Review by davidacevedo — APPROVED** [review]
Looks good 👍

File: package.json (hunk 4 above)
Comment (mdalpozzo) [reply package.json:89]: 👍  done
